### PR TITLE
DAOS-10081 control: Make pool arg non-optional

### PR DESCRIPTION
**Pull request by mjmac**
This should have been done when the old --pool flag was
removed. Also implements DAOS-7964, which was a TODO to
remove the --name/--value flags from the pool set-prop
command. Finally, removes the deprecated --label flag for
pool create, and makes the label a required argument.

Required-githooks: true

Signed-off-by: Michael MacDonald <mjmac.macdonald@intel.com>

### CODE REVIEW

**Review by daltonbohning — COMMENTED** [review]
ftest LGTM

**Comment by mjmac** [comment]
NB: This did run with `Features: pool` on [#3](https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-11251/3/pipeline), but I forgot to re-add the pragma for the final run.

**Review by tanabarr — DISMISSED** [review]

File: src/control/cmd/dmg/pool_test.go [hunk 1]
```diff
@@ -133,145 +127,120 @@ func TestPoolCommands(t *testing.T) {
...
-						propWithVal("label", "foo"),
-					},
-				}),
-			}, " "),
-			nil,
-		},
 		{
 			"Create pool with missing arguments",
-			"pool create",
+			"pool create label",
```
[the remark this answers is not in the document]
Comment (mjmac) [reply src/control/cmd/dmg/pool_test.go:132]: Done, thanks.

**Review by kjacque — APPROVED** [review]

**Review by tanabarr — APPROVED** [review]

**Review by knard-intel — APPROVED** [review]
LGTM.
Good thing that you fixed my horrible workaround for checking if tierratio has been set or not :)

**Pull request merged**